packages/react-core/src/components/Alert/Alert.tsx

```diff
@@ -91,12 +96,57 @@ export const Alert: React.FunctionComponent<AlertProps> = ({
       setIsTooltipVisible(showTooltip);
     }
   }, [titleRef, truncateTitle, isTooltipVisible]);
-  const customClassName = css(
-    styles.alert,
-    isInline && styles.modifiers.inline,
-    variant !== AlertVariant.default && styles.modifiers[variant as 'success' | 'danger' | 'warning' | 'info'],
-    className
-  );
+
+  const [timedOut, setTimedOut] = useState(false);
+  const [timedOutAnimation, setTimedOutAnimation] = useState(true);
+  const [isMouseOver, setIsMouseOver] = useState<boolean | undefined>();
+  const [containsFocus, setContainsFocus] = useState<boolean | undefined>();
```

Comment:
```suggestion
  const [isMouseOver, setIsMouseOver] = useState<boolean>(false);
  const [containsFocus, setContainsFocus] = useState<boolean>(false);
```

Reply:
This won't work because then this `useEffect`'s condition will needlessly happen on first render:
```
  React.useEffect(() => {
    if (containsFocus === false || isMouseOver === false) {
      const timer = setTimeout(() => setTimedOutAnimation(true), timeoutAnimation);
      return () => clearTimeout(timer);
    }
  }, [containsFocus, isMouseOver]);
```

Reply:
tricky tri-state booleans!